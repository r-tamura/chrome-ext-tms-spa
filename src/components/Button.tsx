import * as React from "react"
import classNames from "~/helpers/classNames"
import { Link } from "react-router-dom"

enum Color {
  DEFAULT          = "",
  PRIMARY          = "primary",
  SECONDARY        = "secondary",
  DANGER           = "danger",
  LINK             = "link",
}

interface IButtonProps extends React.ButtonHTMLAttributes<{}> {
  /* button要素とするかa要素とするか true: button / false: a*/
  button?: boolean
  className?: string
  color?: Color
  /* ボタンをblock要素とするか コンテナコンポーネントに対して最大幅と同じになる */
  block?: boolean
  circle?: boolean
  component?: React.AllHTMLAttributes<{}>
  to?: string
  size?: string
}

const Button: React.SFC<IButtonProps> = ({
  button        = true,
  disabled      = false,
  color         = Color.DEFAULT,
  block         = false,
  circle        = false,
  className: classNameProps,
  component,
  children,
  ...otherProps
} = {}) => {
  const className = classNames(
    button ? "tms-btn" : null,
    color || Color.DEFAULT,
    { disabled, block, circle },
    classNameProps
  )
  const Component = component || button ? "button" : Link

  return (
    <Component className={className} {...otherProps}>
      {children}
    </Component>
  )
}

export {
  IButtonProps,
  Color,
  Button,
}

export default Button
