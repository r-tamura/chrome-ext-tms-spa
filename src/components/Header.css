:root {
  --header-height: 50px;
}

.company-logo {
  display: block;
  width: 40px;
}


.page-header {
  display: flex;
  align-items: center;
  flex-direction: row;
  padding: 0 5px;
  background-color: var(--primary-color-reverse);
  /* color: var(--white); */
  font-size: var(--font-size-lg);
  font-weight: bolder;
  line-height: var(--header-height);
  
  @apply --z-index-high;
  @apply --box-shadow;

  & a:focus, & a:hover {
    opacity: 0.8
  }

  & .nav {
    height: 80%;
    margin: 5px 0;
  }

  & .nav-link {
    height: 100%;

    &.active {
      background-color: $nav-selected-bg-color;
    }
  }

  @media (--md) {
    /* ロゴ非表示 */
    .nav, .nav-link {
        display: none;
    }

    .menu-btn {
        display: block;
    }
  }
}

/* ヘッダーメニューボタン */
.menu-btn {
    cursor: pointer;
    padding: $header-height/4 10px;
    display: none;
    &:hover {
      background-color: darken($COLOR_HEADER_BG, 10%);
      border-radius: 50%;
    }

    img {
      height: 50%;
    }
}

.header-user {
  /* @extend .clickable; */
  display: flex;
  align-items: center;
  margin-left: auto;

  .person-icon {
    background-image: url(chrome-extension://odbcdehjjdjjnpenjbblpmkkibggblpe/chrome/assets/images/person.svg);
    height: 50%;
    width: 25px;
    display: block;
  }

  &:hover {
    text-decoration: underline;
  }
}

.header-user, .nav-link {
  img {
    display: inline-block;
    height: $header-height;
    width: $header-height;
    padding: $header-height * 0.10 10px;
  }
}

.header-user .dropdown-menu {
  right: 0px;
}
