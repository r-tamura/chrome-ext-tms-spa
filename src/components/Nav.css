.side-nav {
  background-color: var(--nav-bg-color);
  @apply --z-index-mid;
  @apply --box-shadow;
}

.nav-item {
  color: var(--nav-item-color);
  background-color: inherit;

  &.selected {
    background-color: var(--nav-item-bg-color-selected);

    &:hover {
      opacity: 1;
    }
  }

  &:visited {
    color: var(--nav-item-color);
  }

  &.disabled, &:disabled, &.disabled:hover, &:disabled:hover {
    opacity: 1;
    color: var(--nav-item-color-disabled);
    background-color: inherit;
    pointer-events: none;
  }
}
