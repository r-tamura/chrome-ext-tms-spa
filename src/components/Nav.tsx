import * as React from "react"
import { Link, RouteComponentProps } from "react-router-dom"
import NavItem from "~/components/NavItem"

interface IProps extends React.Props<{}> {
  path: string
}

const Nav: React.SFC<IProps> = ({path}) => {
  const pathname = path.replace(/^\/([^/]+).*$/, "$1")
  return (
    <nav id="side-nav-bar" className="side-nav">
      <ul>
        {/* リンク一覧 */}
        <li>
          <NavItem
            to={"transportation"}
            selected={"transportation" === pathname}
          >
          交通費
          </NavItem>
        </li>
        <li>
          <NavItem
            to={"attendance"}
            selected={"attendance" === pathname}
          >
          勤怠管理
          </NavItem>
        </li>
        <li>
          <NavItem
            to={"resource"}
            disabled={true}
            selected={"resource" === pathname}
          >
          資源管理
          </NavItem>
        </li>
      </ul>
    </nav>
  )
}

export default Nav
