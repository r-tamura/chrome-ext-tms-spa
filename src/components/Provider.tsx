import React from "react";
import { ThemeProvider } from "styled-components";
import { telema } from "~/styles/theme";

export const AppThemeProvider: React.SFC<{ children: JSX.Element }> = ({
  children
}) => <ThemeProvider theme={telema}>{children}</ThemeProvider>;
