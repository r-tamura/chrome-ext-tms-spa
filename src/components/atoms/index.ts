export * from "./Button";
export * from "./Panel";
export * from "./Modal";
export * from "./Form";
export * from "./TextField";
