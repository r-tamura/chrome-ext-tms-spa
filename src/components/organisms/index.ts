export * from "./AppGrid";
export * from "./MainContent";
export * from "./Header";
export * from "./TransExpenseHistory";
export * from "./TransExpenseTemplates";
export * from "./Nav";
