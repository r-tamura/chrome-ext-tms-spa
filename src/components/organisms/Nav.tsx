import React from "react";
import { NavItem } from "~/components/molecules";
import styled from "styled-components";
import { MID } from "~/styles/zIndex";
import { STANDARD } from "~/styles/boxShadow";
import { ThemeProps } from "~/styles/theme";

interface INavProps extends React.Props<{}> {
  path: string;
}

export const Nav: React.SFC<INavProps> = ({ path }) => {
  const pathname = path.replace(/^\/([^/]+).*$/, "$1");
  return (
    <NavStyled id="side-nav-bar">
      <ul>
        {/* リンク一覧 */}
        <li>
          <NavItem
            to={"transportation"}
            selected={"transportation" === pathname}
          >
            交通費
          </NavItem>
        </li>
        <li>
          <NavItem to={"attendance"} selected={"attendance" === pathname}>
            勤怠管理
          </NavItem>
        </li>
        {/* TODO: 資源管理は未実装 */}
        <li>
          <NavItem
            to={"resource"}
            disabled={true}
            selected={"resource" === pathname}
          >
            資源管理
          </NavItem>
        </li>
      </ul>
    </NavStyled>
  );
};

const NavStyled = styled.nav<ThemeProps>`
  background-color: ${({ theme }) => theme.primary};
  z-index: ${MID};
  box-shadow: ${STANDARD};
`;
