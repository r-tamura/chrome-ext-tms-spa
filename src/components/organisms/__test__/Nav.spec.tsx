import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from "react-router";
import Nav, { INavProps } from "~/components/organisms/Nav";
import { AppThemeProvider } from "~/components/Provider";

const createComponent = (props: INavProps = { path: "transportation" }) => (
  <AppThemeProvider>
    <MemoryRouter>
      <Nav {...props}>Item</Nav>
    </MemoryRouter>
  </AppThemeProvider>
);

describe("<Nav />", () => {
  it("Snapshot", () => {
    const $ = mount(createComponent({ path: "transportation" }));
    expect(toJson($)).toMatchSnapshot();
  });

  describe("交通費選択中", () => {
    const $ = mount(createComponent({ path: "transportation" }));
    const $NavItems = $.find("NavItem");

    it("リンクは3つ", () => {
      expect($NavItems.length).toBe(3);
    });

    it("交通費が選択されている", () => {
      expect($NavItems.at(0).text()).toBe("交通費");
      expect($NavItems.at(0).prop("selected")).toBeTruthy();
    });

    it("勤怠管理が選択されていない", () => {
      expect($NavItems.at(1).text()).toBe("勤怠管理");
      expect($NavItems.at(1).prop("selected")).toBeFalsy();
    });

    it("資源管理は未実装のため無効", () => {
      expect($NavItems.at(2).prop("disabled")).toBeTruthy();
    });
  });

  describe("勤怠管理選択中", () => {
    const w = mount(createComponent({ path: "attendance" }));
    const $NavItems = w.find("NavItem");

    it("交通費管理が選択されていない", () => {
      expect($NavItems.at(0).prop("selected")).toBeFalsy();
    });

    it("勤怠管理が選択されている", () => {
      expect($NavItems.at(1).prop("selected")).toBeTruthy();
    });
  });
});
