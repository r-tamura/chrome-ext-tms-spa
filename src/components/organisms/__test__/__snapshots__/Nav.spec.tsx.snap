// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> Snapshot 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "black": "rgb(5, 5, 5)",
        "buttonDefault": "rgba(0, 0, 0, 0.08)",
        "danger": "#D32F2F",
        "fontFamily": "\\"M PLUS Rounded 1c\\", sans-serif",
        "formBorderColor": "rgba(0, 0, 0, 0.16)",
        "formBorderColorDisabled": "rgba(0, 0, 0, 0.08)",
        "formBorderColorError": "#D32F2F",
        "formBorderColorFocused": "hsl(206, 100%, 31%)",
        "hoveredBg": "rgba(0, 0, 0, 0.03)",
        "inputPlaceHolder": "rgba(0, 0, 0, 0.09)",
        "primary": "hsl(206, 100%, 31%)",
        "primaryDark": "hsl(208, 80%, 27%)",
        "primaryHeavyDark": "hsl(208, 80%, 27%)",
        "primaryHeavyLight": "hsl(199, 91%, 64%);",
        "primaryLight": "hsl(201, 98%, 41%);",
        "primaryReverse": "hsla(0, 0%, 99%, 1);",
        "primaryReverseDark": "hsla(0, 0%, 99%, 0.5);",
        "secondary": "#689F38",
        "textGray": "hsla(0, 5%, 5%, .25)",
        "textMain": "hsl(0, 5%, 5%)",
        "white": "rgb(253, 253, 253)",
      }
    }
  >
    <MemoryRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "iyu6ya",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "iyu6ya",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Component
          path="transportation"
        >
          <styled.nav
            id="side-nav-bar"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-hMqMXs",
                    "isStatic": false,
                    "lastClassName": "hMqybw",
                    "rules": Array [
                      "
  background-color: ",
                      [Function],
                      ";
  z-index: ",
                      "0",
                      ";
  box-shadow: ",
                      "0px 0px 5px #222222",
                      ";
",
                    ],
                  },
                  "displayName": "styled.nav",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-hMqMXs",
                  "target": "nav",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="side-nav-bar"
            >
              <nav
                className="sc-hMqMXs hMqybw"
                id="side-nav-bar"
              >
                <ul>
                  <li>
                    <WithTheme(Component)
                      selected={true}
                      to="transportation"
                    >
                      <Component
                        selected={true}
                        theme={
                          Object {
                            "black": "rgb(5, 5, 5)",
                            "buttonDefault": "rgba(0, 0, 0, 0.08)",
                            "danger": "#D32F2F",
                            "fontFamily": "\\"M PLUS Rounded 1c\\", sans-serif",
                            "formBorderColor": "rgba(0, 0, 0, 0.16)",
                            "formBorderColorDisabled": "rgba(0, 0, 0, 0.08)",
                            "formBorderColorError": "#D32F2F",
                            "formBorderColorFocused": "hsl(206, 100%, 31%)",
                            "hoveredBg": "rgba(0, 0, 0, 0.03)",
                            "inputPlaceHolder": "rgba(0, 0, 0, 0.09)",
                            "primary": "hsl(206, 100%, 31%)",
                            "primaryDark": "hsl(208, 80%, 27%)",
                            "primaryHeavyDark": "hsl(208, 80%, 27%)",
                            "primaryHeavyLight": "hsl(199, 91%, 64%);",
                            "primaryLight": "hsl(201, 98%, 41%);",
                            "primaryReverse": "hsla(0, 0%, 99%, 1);",
                            "primaryReverseDark": "hsla(0, 0%, 99%, 0.5);",
                            "secondary": "#689F38",
                            "textGray": "hsla(0, 5%, 5%, .25)",
                            "textMain": "hsl(0, 5%, 5%)",
                            "white": "rgb(253, 253, 253)",
                          }
                        }
                        to="transportation"
                      >
                        <MenuItem
                          disabled={false}
                          style={
                            Object {
                              "backgroundColor": "hsl(208, 80%, 27%)",
                            }
                          }
                        >
                          <styled.div
                            as="div"
                            disabled={false}
                            style={
                              Object {
                                "backgroundColor": "hsl(208, 80%, 27%)",
                              }
                            }
                          >
                            <StyledComponent
                              as="div"
                              disabled={false}
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-ifAKCX",
                                    "isStatic": false,
                                    "lastClassName": "bcZfdK",
                                    "rules": Array [
                                      "
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 12px;
  padding-bottom: 12px;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5em;
  white-space: nowrap;
  text-overflow: ellipsis;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }

  ",
                                      [Function],
                                      ";
",
                                    ],
                                  },
                                  "displayName": "styled.div",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-ifAKCX",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              style={
                                Object {
                                  "backgroundColor": "hsl(208, 80%, 27%)",
                                }
                              }
                            >
                              <div
                                className="sc-ifAKCX blmhNe"
                                disabled={false}
                                style={
                                  Object {
                                    "backgroundColor": "hsl(208, 80%, 27%)",
                                  }
                                }
                              >
                                <Styled(Link)
                                  to="transportation"
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "sc-EHOje",
                                          "isStatic": false,
                                          "lastClassName": "jKfzqM",
                                          "rules": Array [
                                            "
  color: ",
                                            [Function],
                                            ";
  display: block;
",
                                          ],
                                        },
                                        "displayName": "Styled(Link)",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "sc-EHOje",
                                        "target": [Function],
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    to="transportation"
                                  >
                                    <Link
                                      className="sc-EHOje jKfzqM"
                                      to="transportation"
                                    >
                                      <a
                                        className="sc-EHOje jKfzqM"
                                        href="/transportation"
                                        onClick={[Function]}
                                      >
                                        交通費
                                      </a>
                                    </Link>
                                  </StyledComponent>
                                </Styled(Link)>
                              </div>
                            </StyledComponent>
                          </styled.div>
                        </MenuItem>
                      </Component>
                    </WithTheme(Component)>
                  </li>
                  <li>
                    <WithTheme(Component)
                      selected={false}
                      to="attendance"
                    >
                      <Component
                        selected={false}
                        theme={
                          Object {
                            "black": "rgb(5, 5, 5)",
                            "buttonDefault": "rgba(0, 0, 0, 0.08)",
                            "danger": "#D32F2F",
                            "fontFamily": "\\"M PLUS Rounded 1c\\", sans-serif",
                            "formBorderColor": "rgba(0, 0, 0, 0.16)",
                            "formBorderColorDisabled": "rgba(0, 0, 0, 0.08)",
                            "formBorderColorError": "#D32F2F",
                            "formBorderColorFocused": "hsl(206, 100%, 31%)",
                            "hoveredBg": "rgba(0, 0, 0, 0.03)",
                            "inputPlaceHolder": "rgba(0, 0, 0, 0.09)",
                            "primary": "hsl(206, 100%, 31%)",
                            "primaryDark": "hsl(208, 80%, 27%)",
                            "primaryHeavyDark": "hsl(208, 80%, 27%)",
                            "primaryHeavyLight": "hsl(199, 91%, 64%);",
                            "primaryLight": "hsl(201, 98%, 41%);",
                            "primaryReverse": "hsla(0, 0%, 99%, 1);",
                            "primaryReverseDark": "hsla(0, 0%, 99%, 0.5);",
                            "secondary": "#689F38",
                            "textGray": "hsla(0, 5%, 5%, .25)",
                            "textMain": "hsl(0, 5%, 5%)",
                            "white": "rgb(253, 253, 253)",
                          }
                        }
                        to="attendance"
                      >
                        <MenuItem
                          disabled={false}
                          style={null}
                        >
                          <styled.div
                            as="div"
                            disabled={false}
                            style={null}
                          >
                            <StyledComponent
                              as="div"
                              disabled={false}
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-ifAKCX",
                                    "isStatic": false,
                                    "lastClassName": "bcZfdK",
                                    "rules": Array [
                                      "
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 12px;
  padding-bottom: 12px;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5em;
  white-space: nowrap;
  text-overflow: ellipsis;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }

  ",
                                      [Function],
                                      ";
",
                                    ],
                                  },
                                  "displayName": "styled.div",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-ifAKCX",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              style={null}
                            >
                              <div
                                className="sc-ifAKCX blmhNe"
                                disabled={false}
                                style={null}
                              >
                                <Styled(Link)
                                  to="attendance"
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "sc-EHOje",
                                          "isStatic": false,
                                          "lastClassName": "jKfzqM",
                                          "rules": Array [
                                            "
  color: ",
                                            [Function],
                                            ";
  display: block;
",
                                          ],
                                        },
                                        "displayName": "Styled(Link)",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "sc-EHOje",
                                        "target": [Function],
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    to="attendance"
                                  >
                                    <Link
                                      className="sc-EHOje jKfzqM"
                                      to="attendance"
                                    >
                                      <a
                                        className="sc-EHOje jKfzqM"
                                        href="/attendance"
                                        onClick={[Function]}
                                      >
                                        勤怠管理
                                      </a>
                                    </Link>
                                  </StyledComponent>
                                </Styled(Link)>
                              </div>
                            </StyledComponent>
                          </styled.div>
                        </MenuItem>
                      </Component>
                    </WithTheme(Component)>
                  </li>
                  <li>
                    <WithTheme(Component)
                      disabled={true}
                      selected={false}
                      to="resource"
                    >
                      <Component
                        disabled={true}
                        selected={false}
                        theme={
                          Object {
                            "black": "rgb(5, 5, 5)",
                            "buttonDefault": "rgba(0, 0, 0, 0.08)",
                            "danger": "#D32F2F",
                            "fontFamily": "\\"M PLUS Rounded 1c\\", sans-serif",
                            "formBorderColor": "rgba(0, 0, 0, 0.16)",
                            "formBorderColorDisabled": "rgba(0, 0, 0, 0.08)",
                            "formBorderColorError": "#D32F2F",
                            "formBorderColorFocused": "hsl(206, 100%, 31%)",
                            "hoveredBg": "rgba(0, 0, 0, 0.03)",
                            "inputPlaceHolder": "rgba(0, 0, 0, 0.09)",
                            "primary": "hsl(206, 100%, 31%)",
                            "primaryDark": "hsl(208, 80%, 27%)",
                            "primaryHeavyDark": "hsl(208, 80%, 27%)",
                            "primaryHeavyLight": "hsl(199, 91%, 64%);",
                            "primaryLight": "hsl(201, 98%, 41%);",
                            "primaryReverse": "hsla(0, 0%, 99%, 1);",
                            "primaryReverseDark": "hsla(0, 0%, 99%, 0.5);",
                            "secondary": "#689F38",
                            "textGray": "hsla(0, 5%, 5%, .25)",
                            "textMain": "hsl(0, 5%, 5%)",
                            "white": "rgb(253, 253, 253)",
                          }
                        }
                        to="resource"
                      >
                        <MenuItem
                          disabled={true}
                          style={null}
                        >
                          <styled.div
                            as="div"
                            disabled={true}
                            style={null}
                          >
                            <StyledComponent
                              as="div"
                              disabled={true}
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "sc-ifAKCX",
                                    "isStatic": false,
                                    "lastClassName": "bcZfdK",
                                    "rules": Array [
                                      "
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 12px;
  padding-bottom: 12px;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5em;
  white-space: nowrap;
  text-overflow: ellipsis;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }

  ",
                                      [Function],
                                      ";
",
                                    ],
                                  },
                                  "displayName": "styled.div",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "sc-ifAKCX",
                                  "target": "div",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                              style={null}
                            >
                              <div
                                className="sc-ifAKCX bcZfdK"
                                disabled={true}
                                style={null}
                              >
                                <styled.div
                                  to="resource"
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "sc-bZQynM",
                                          "isStatic": false,
                                          "lastClassName": "cPlydf",
                                          "rules": Array [
                                            "
  color: ",
                                            [Function],
                                            ";
  opacity: 0.7;
",
                                          ],
                                        },
                                        "displayName": "styled.div",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "sc-bZQynM",
                                        "target": "div",
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    to="resource"
                                  >
                                    <div
                                      className="sc-bZQynM cPlydf"
                                      to="resource"
                                    >
                                      資源管理
                                    </div>
                                  </StyledComponent>
                                </styled.div>
                              </div>
                            </StyledComponent>
                          </styled.div>
                        </MenuItem>
                      </Component>
                    </WithTheme(Component)>
                  </li>
                </ul>
              </nav>
            </StyledComponent>
          </styled.nav>
        </Component>
      </Router>
    </MemoryRouter>
  </ThemeProvider>
</Component>
`;
